File: ios/Calculator.m
//
//  Calculator.m
//  NativeModuleDemo
//
//  Created by Bhargav Lakkur Kusha Kumar on 12/27/24.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE (Calculator, NSObject)
RCT_EXTERN_METHOD(add
                  : (nonnull NSNumber *)a b
                  : (nonnull NSNumber *)b callback
                  : (RCTResponseSenderBlock)callback)
@end

File: ios/Calculator.swift
//
//  Calculator.swift
//  NativeModuleDemo
//
//  Created by Bhargav Lakkur Kusha Kumar on 12/27/24.
//

import Foundation
import React

@objc(Calculator)
class Calculator: NSObject {

    @objc
    func add(_ a: NSNumber, b: NSNumber, callback: @escaping RCTResponseSenderBlock) {
        let sum = a.doubleValue + b.doubleValue
        callback([NSNull(), sum])  // Sending the result back to JavaScript
    }

    static func requiresMainQueueSetup() -> Bool {
        return false
    }
}
